src/infer_response.h

```diff
@@ -118,10 +119,18 @@ class InferenceResponseFactory {
   std::function<void(std::unique_ptr<InferenceResponse>&&, const uint32_t)>
       response_delegator_;
 
+
 #ifdef TRITON_ENABLE_TRACING
   // Inference trace associated with this response.
   std::shared_ptr<InferenceTraceProxy> trace_;
 #endif  // TRITON_ENABLE_TRACING
+
+  // The internal unique ID of the request
+  uint64_t request_id_;
```

Comment:
How does `request_id_` differ from `id_` ?
```
  // The ID of the corresponding request that should be included in
  // every response.
  std::string id_;
```

Maybe the comments should be updated to make them more clearly distinct

Reply:
The id_ is a property that can be optionally provided by the user. request_id_ is a unique identification that triton attaches to the request to make this feature work for cases where user doesnot provide id_. 
I agree it is little confusing within this header. 
@Tabrizian  We can elaborate this in L102-L103 about the nature of id_ field.

Reply:
Updated.